File: command/python/stock_controller.py
from order_command import OrderCommand


class StockController:
    """
    Command (IOrderCommand) arayüzünü uygulayan sınıf referansını tutar.
    Metotun ne zaman çağrılacağını belirtir.
    UML diyagramındaki Invoker sınıfına denk gelmektedir.
    """

    def __init__(self):
        self._order_commands = []

    def take_order(self, command: OrderCommand):
        self._order_commands.append(command)

    def place_orders(self):
        for command in self._order_commands:
            command.execute()

File: command/typescript/stock-controller.ts
import {IOrderCommand} from "./order-command.interface";

// Command (IOrderCommand) arayüzünü uygulayan sınıf referansını tutar.
// Metotun ne zaman çağrılacağını belirtir.
// UML diyagramındaki Invoker sınıfına denk gelmektedir.
export class StockController {
  private orderCommands: IOrderCommand[] = [];

  public takeOrder(command: IOrderCommand): void {
    this.orderCommands.push(command);
  }

  public placeOrders() {
    for (let command of this.orderCommands) {
      command.execute();
    }

    this.orderCommands = [];
  }
}
